src/_pytest/config/__init__.py

```diff
@@ -470,9 +470,18 @@ def _importconftest(self, conftestpath):
     #
 
     def consider_preparse(self, args):
-        for opt1, opt2 in zip(args, args[1:]):
-            if opt1 == "-p":
-                self.consider_pluginarg(opt2)
+        i = 0
```

Comment:
i believe a ```if opt1.startswith("-p"): self.consider_pluginarg(opt1[2:])``` might be neater

Reply:
opt1 might be a py.LocalPath.

Reply:
ah, true, a str call may be needed then

Reply:
In the long run we'll have a argparse namespace there - had this today already, and then it can be much simpler.  Will leave it for now as-is.